src/lbaf/Model/lbsPhase.py

```diff
@@ -406,6 +408,151 @@ def populate_from_log(self, phase_id):
+
+        ranks: Dict[int, Rank] = {
+            r_id: Rank(self.__logger, r_id) for r_id in spec.get("ranks", []).keys()}
+        objects: Dict[int, Object] = {}  # object instances (indexed by id)
+        shared_blocks: Dict[int, Block] = {}  # shared blocks instances (indexed by id)
+
+        if len(spec["ranks"].keys()) == 0:
+            raise RuntimeError("Missing rank distributions")
+
+        # Load objects set
+        for rank_id, rank_specs in spec["ranks"].items():
+            task_ids = rank_specs["tasks"]
+            for task_id in task_ids:
+                task_user_defined = {}
+                time = spec["tasks"][task_id]
+
+                # Check: task cannot reside in more than 1 rank at the same time
+                if task_id in objects:
+                    raise RuntimeError(
+                        f"Cannot assign task {task_id} to rank {rank_id}: "
+                        f"already assigned to rank {objects[task_id].get_rank_id()}"
+                    )
+
+                o = Object(
+                    task_id,
+                    r_id=rank_id,
+                    load=time,
+                    user_defined=task_user_defined)
+
+                objects[task_id] = o
+
+                # Set migratable
+                ranks[rank_id].add_migratable_object(o)
+
+        # Load shared blocks
+        for shared_id, shared_block_spec in spec["shared_blocks"].items():
+            if not shared_block_spec["tasks"]:
+                self.__logger.warning(f"No tasks linked to shared block {shared_id} !")
+
+            # Browse tasks
+            for task_id in shared_block_spec["tasks"]:
+                o = objects[task_id]
+
+                # Find shared block and create if not already created in memory
+                b: Block = None
+                if not shared_id in shared_blocks:
+                    b = Block(b_id=shared_id, h_id=shared_block_spec["home"], size=shared_block_spec["size"],
+                              o_ids=shared_block_spec["tasks"])
+                    # Index the shared block for next loops checks
+                    shared_blocks[shared_id] = b
+                else:
+                    b = shared_blocks[shared_id]
+
+                # Associate shared block with object
+                # Check: task must share 0 or 1 block
+                if not multiple_sharing and o.get_shared_block_id() is not None and o.get_shared_block_id() != shared_id:
+                    raise RuntimeError(
+                        f"Task {o.get_id()} already shared block {o.get_shared_block_id()} and cannot share additional block {shared_id}. Only 0 or 1 allowed")
+                o.set_shared_block(b)
+
+                # Initialize object user defined data
+                o.get_user_defined()["shared_id"] = b.get_id()
+                o.get_user_defined()["shared_bytes"] = b.get_size()
+                o.get_user_defined()["home"] = b.get_home_id()
```

Comment:
I was remembering wrong. This should be `home_rank`, not `home`, per https://github.com/DARMA-tasking/vt/blob/bc5264ab35da2a303c4db3d290dccc00eca67c83/src/vt/vrt/collection/balance/temperedlb/temperedlb.cc#L597. This change might need to be made in other places as well.

Reply:
@ppebay

Reply:
@tlamonthezie please fix accordingly, thanks